core/lib/resources/presence/sentry.js

```diff
@@ -1,159 +1,270 @@
-var logging = require('minilog')('radar:presence:sentry'),
-    Persistence = require('persistence');
+var _ = require('underscore'),
+    Minilog = require('minilog'),
+    logging = Minilog('radar:sentry'),
+    Persistence = require('persistence'),
+    redisSentriesKey = 'sentry:/radar';
 
-function parse(message) {
-  try {
-    return JSON.parse(message);
+var defaultOptions = {
+  DEFAULT_EXPIRY_OFFSET: 60000,
+  REFRESH_INTERVAL: 10000,
+  CHECK_INTERVAL: 30000
+};
+
+var parseJSON = function(message) {
+  try { 
+    return JSON.parse(message); 
   } catch (e) { }
-}
+};
+
+var messageIsExpired = function(message) {
+  return (!message || !message.expiration || (message.expiration <= Date.now()));
+};
 
-function Sentry(name) {
-  this.name = name;
+var messageExpiration = function(message) {
```

Comment:
probably expand to sentryExpirationMessage and include the `var text = expiration ? expiration + '/' + this._defaultExpiryOffset : 'not-present';` in here.